File: BinarySearch/Python/find_the_distance_value_between_two_arrays.py
class Solution:
    def findTheDistanceValue(self, arr1: list[int], arr2: list[int], d: int) -> int:
        arr2.sort()
        res, n = 0, len(arr2)
        for x in arr1:
            l, r, close = 0, n - 1, False
            while (l <= r) :
                mid = l + (r - l) // 2
                cur = arr2[mid]
                if abs(cur - x) <= d:
                    close = True
                    break
                elif cur < x: l = mid + 1
                else: r = mid - 1
            if not close: res += 1
        return res

File: BinarySearch/CPP/find_the_distance_value_between_two_arrays.cpp
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;
class Solution {
public:
    int findTheDistanceValue(vector<int>& arr1, vector<int>& arr2, int d) {
        sort(arr2.begin(), arr2.end());
        int i = 0, j = 0, m = arr1.size(), n = arr2.size(), res = 0;
        for (int x : arr1) {
            int l = 0, r = n-1;
            bool hasclosevalue = false;
            while( l <= r) {
                int mid = l + (r-l) / 2, cur = arr2[mid];
                if (abs(cur - x) <= d) {
                    hasclosevalue = true;
                    break;
                }
                else if (cur < x) l = mid + 1;
                else r = mid - 1;
            }
            if (!hasclosevalue) res++;
        }
        return res;
    }
};